### ISSUE TITLE
Feature: <ALL> in Mod Selector

### ISSUE THREAD

**Issue by ech3lon42** [Xinaths/QuickArmorRebalance#1]
I hope it's okay to open a feature request like this.

QAR already provides the `<Currently Worn Armor>` selector in the Mod selection Dropdown.

Would it be possible to also add an `<ALL>` selector, which simply lists all found armor pieces (or weapons) and lets us use the awesome and already existing filter function to filter through the entire list of armors/weapons?

I've tried to add this myself and managed to build the sources, however I'm a bit lost when it comes to C++ to be honest.

Something like this is what I had in mind, but I'm not quite sure how/where to add the proper UI Elements:

```
auto dataHandler = RE::TESDataHandler::GetSingleton();
for (auto i : dataHandler->GetFormArray<RE::TESObjectARMO>()) {
    if (!IsValidItem(i)) continue;
    if (bFilterModified && (g_Data.modifiedItems.contains(i) || g_Data.modifiedItemsShared.contains(i)))
        continue;
    if (*itemFilter && !StringContainsI(i->GetFullName(), itemFilter)) continue;

    params.items.push_back(i);
}
```

The ability to filter by slot would also be amazing, e.g. by including checkboxes for the usual slotmasks.

![feature](https://github.com/user-attachments/assets/d72912a6-b00d-41ec-99b9-85c98f8affa2)

**Comment by Xinaths** [comment]
Adding it is fairly simple, but I feel it introduces a lot of potential problems from misuse:

- Performance of various things when unfiltered (even filtering might lag the UI badly)
- Most things expect to only be working within a relatively sane set of items, being able to pass everything in would be an issue
- Item distribution doesn't work for things across mods (and is disabled for current armor set).
- Probably various issues I haven't even thought of

All that said, I'm not super opposed to adding it as a "Use at your own risk" option.

As for filtering by slots, it is a good idea that I will add in the next version.

**Comment by ech3lon42** [comment]
You are right about everything. Although the performance drop is not too bad for me, even with (probably) thousands of armor sets.

I think this has so much potential to serve as a replacement for AddItemMenu and Equipment Managers. I just couldn't ignore that. It's really cool to be able to filter for certain slots (or exclude items that match certain slots). It's a huge time saver to search through all mods quickly. AddItemMenu is very slow in comparison.

But I do understand if you wouldn't want to include it as part of this particular mod.

Anyway, my attempt to contribute something:
https://github.com/Xinaths/QuickArmorRebalance/pull/2
